config/config_source.go

```diff
@@ -1,9 +1,14 @@
 package config
 
 import (
+	"crypto/sha256"
+	"encoding/hex"
+	"io/ioutil"
 	"sync"
 
 	"github.com/fsnotify/fsnotify"
```

Comment:
unrelated, but would be nice to standardize on a single notify package here.

Reply:
We would have to fork viper, or not use its change detection to do this.

Reply:
Ah okay, this is fine then.